.github/workflows/feature-branch-deployment.yaml

```diff
@@ -0,0 +1,46 @@
+name: Feature Branch Deployment
+
+on:
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check branch name
+        run: |
+          if [[ "${GITHUB_REF_NAME}" == "main" ]] || ! [[ "${GITHUB_REF_NAME}" =~ ^[0-9]+-.* ]]; then
```

Comment:
As `main` does not match the regex defined for the feature branch name pattern, you may be able to simplify this condition by removing the first operand:

`if ! [[ "${GITHUB_REF_NAME}" =~ ^[0-9]+-.* ]]; then`

Reply:
You can use the `on.push.branches` trigger with branch name patterns set